examples/colormap_normalization/colormap_normalizations_piecewisenorm.py

```diff
@@ -0,0 +1,103 @@
+"""
+============================================
+Examples of arbitrary colormap normalization
+============================================
+
+Here I plot an image array with data spanning for a large dynamic range,
+using different normalizations. Look at how each of them enhances
+different features.
+
+"""
+
+
+import matplotlib.cm as cm
+import matplotlib.colors as colors
+import matplotlib.pyplot as plt
+
+from mpl_toolkits.mplot3d import Axes3D
+
+import numpy as np
+
+from sampledata import PiecewiseNormData
+
+X, Y, data = PiecewiseNormData()
+cmap = cm.spectral
+
+# Creating functions for plotting
+
+
+def make_plot(norm, label=''):
```

Comment:
Is there anyway humanly possible to reduce this example to just being about the norms? Something like:
```
fig, (ax1, ax2, ax3, ax4) = plt.subplots(2,2)
 data = something easy and simple

pnorm = ...
im = ax1.imshow(data, norm=pnorm)
cb = ax1.colorbar(im)
cb.set_title("PiecewiseNorm")

mnorm = ...
ax2.imshow(data, norm=mnorm)
 etc...
```

I think the important piece (the new norm) gets swamped under all this boilerplate prettification stuff. And I think the 3D should be its own example.